egs/wsj/s5/steps/cleanup/create_segments_from_ctm.pl

```diff
@@ -325,7 +325,7 @@ sub ProcessWav {
   # Save the aligned CTM if needed
   if(defined($ACT)){
     for (my $i=0; $i<=$#aligned_ctm; $i++) {
```

Comment:
I recall a segment file has the following fields (at least in some old
scripts?):

"utterance-id recording-id start-time end-time"

In this script, I was creating segment file using:

171   print $SO "$seg_id $wav_id $seg_start $seg_end\n";

So I probably messed it up in the first place. I had a look at the calling
scripts, In wsj/s5/local/run_segmentation.sh, steps/get_ctm.sh is called
with --use-segments=true (default value), and it uses a
"fake" reco2file_and_channel that I create in this
script steps/cleanup/split_long_utterance.sh using the following lines:

# CAVEAT: We are not dealing with channels here. Each channel should have a
# unique file name in wav.scp.
paste -d ' ' <(cut -d ' ' -f 1 $output_dir/wav.scp) \
  <(cut -d ' ' -f 1 $output_dir/wav.scp) | awk '{print $1" "$2" A";}' \
  > $output_dir/reco2file_and_channel

Should we fix this?

Guoguo

On Mon, Apr 4, 2016 at 5:20 PM, Daniel Povey <notifications@github.com>
wrote:

> In egs/wsj/s5/steps/cleanup/create_segments_from_ctm.pl
> <https://github.com/kaldi-asr/kaldi/pull/651#discussion_r58470566>:
>
> > @@ -325,7 +325,7 @@ sub ProcessWav {
> >    # Save the aligned CTM if needed
> >    if(defined($ACT)){
> >      for (my $i=0; $i<=$#aligned_ctm; $i++) {
>
> Actually on reviewing the calling script I think it may have been using a
> real wav-id with a channel. See my more recent PR for my proposed fix. I
> realized after writing the email that there are really only two types of
> CTMs in Kaldi: one with utterance-id (and channel = 1), and one with the
> wav-id (aka file) and channel. And I think it's the latter that this one
> uses. Dan
> … <#m_6928870947465807567_>
> On Mon, Apr 4, 2016 at 8:14 PM, Guoguo Chen <notifications@github.com>
> wrote: In egs/wsj/s5/steps/cleanup/create_segments_from_ctm.pl <#651
> (comment)
> <https://github.com/kaldi-asr/kaldi/pull/651#discussion_r58470111>>: > @@
> -325,7 +325,7 @@ sub ProcessWav { > # Save the aligned CTM if needed >
> if(defined($ACT)){ > for (my $i=0; $i<=$#aligned_ctm; $i++) { Sorry just
> saw this. That part of the code was added by Nagendra and he originally
> just wanted to dump some temporary file instead of a real CTM I think. And
> yes the script uses Kaldi format CTM which uses recording-id instead of
> wav-id (it creates a segment file which uses recording-id's). I can make a
> pull request to change the wav_id to recording_id if that's not already
> done. Guoguo … <#m_1570072994657872712_> On Mon, Apr 4, 2016 at 1:18 PM,
> Daniel Povey <notifications@github.com> wrote: In
> egs/wsj/s5/steps/cleanup/create_segments_from_ctm.pl <#651
> <https://github.com/kaldi-asr/kaldi/pull/651> (comment) <#651 (comment)
> <https://github.com/kaldi-asr/kaldi/pull/651#discussion_r58441040>>>: >
> @@ -325,7 +325,7 @@ sub ProcessWav { > # Save the aligned CTM if needed >
> if(defined($ACT)){ > for (my $i=0; $i<=$#aligned_ctm; $i++) { If it's
> called CTM it should be a correctly formatted CTM. It seems the script
> requires the input to be one of those Kaldi-style CTMs where instead of
> "wav-id channel" it has "recording-id 1", or "recording-id A" (they're
> equivalent). This should be documented at the top of the script. It can
> just print out another Kaldi-style ctm, with "recording_id 1" at the start,
> but the wav_id in the script should be changed to say recording_id, because
> that's what it really is. Remi, could you modify the pull request like
> this? Cc'ing Nagendra in case he was relying on this output for something.
> Dan … <#m_-177697443212186889_> On Mon, Apr 4, 2016 at 3:26 PM, Rémi
> Francis <notifications@github.com> wrote: In egs/wsj/s5/steps/cleanup/
> create_segments_from_ctm.pl <#651
> <https://github.com/kaldi-asr/kaldi/pull/651> <#651
> <https://github.com/kaldi-asr/kaldi/pull/651>> (comment) <#651
> <https://github.com/kaldi-asr/kaldi/pull/651> (comment) <#651 (comment)
> <https://github.com/kaldi-asr/kaldi/pull/651#discussion_r58432923>>>>: >
> @@ -325,7 +325,7 @@ sub ProcessWav { > # Save the aligned CTM if needed >
> if(defined($ACT)){ > for (my $i=0; $i<=$#aligned_ctm; $i++) { It currently
> prints " ", where error is 1 if the word was right in the decoded ctm and 0
> otherwise (or the opposite, can't remember). I guess it would be nicer to
> have this a correctly formatted CTM. I just needed the wav id, so I didn't
> bother making it right. — You are receiving this because you commented.
> Reply to this email directly or view it on GitHub <
> https://github.com/kaldi-asr/kaldi/pull/651/files/c90a6034fdf2881f97beaaabab1dd32766125d2c#r58432923>
> — You are receiving this because you were mentioned. Reply to this email
> directly or view it on GitHub <
> https://github.com/kaldi-asr/kaldi/pull/651/files/c90a6034fdf2881f97beaaabab1dd32766125d2c#r58441040>
> — You are receiving this because you modified the open/close state. Reply
> to this email directly or view it on GitHub <
> https://github.com/kaldi-asr/kaldi/pull/651/files/c90a6034fdf2881f97beaaabab1dd32766125d2c#r58470111>
>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/651/files/c90a6034fdf2881f97beaaabab1dd32766125d2c#r58470566>
>